src/sentry_envelope.c

```diff
@@ -400,75 +318,41 @@ prepare_http_request(const sentry_uuid_t *event_id,
 
     h = &rv->headers[rv->headers_len++];
     h->key = "content-type";
-    h->value = sentry__string_clone(content_type);
+    h->value = sentry__string_clone(ENVELOPE_MIME);
 
     h = &rv->headers[rv->headers_len++];
     h->key = "content-length";
     h->value = sentry__int64_to_string((int64_t)payload_len);
 
-    switch (endpoint_type) {
-    case ENDPOINT_TYPE_ENVELOPE:
-        rv->url = sentry__dsn_get_envelope_url(&options->dsn);
-        break;
-    case ENDPOINT_TYPE_STORE:
-        rv->url = sentry__dsn_get_store_url(&options->dsn);
-        break;
-    case ENDPOINT_TYPE_MINIDUMP:
-        rv->url = sentry__dsn_get_minidump_url(&options->dsn);
-        break;
-    case ENDPOINT_TYPE_ATTACHMENT: {
-        rv->url = sentry__dsn_get_attachment_url(&options->dsn, event_id);
-        break;
-    }
-    default:
-        rv->url = NULL;
-    }
-
-    rv->payload = payload;
-    rv->payload_len = payload_len;
-    rv->payload_owned = payload_owned;
+    rv->body = body;
+    rv->body_len = payload_len;
+    rv->body_owned = body_owned;
 
     return rv;
 }
 
-static void
-gen_boundary(char *boundary)
-{
-#if SENTRY_UNITTEST
-    sentry_uuid_t boundary_id
-        = sentry_uuid_from_string("0220b54a-d050-42ef-954a-ac481dc924db");
-#else
-    sentry_uuid_t boundary_id = sentry_uuid_new_v4();
-#endif
-    sentry_uuid_as_string(&boundary_id, boundary);
-    strcat(boundary, "-boundary-");
-}
-
 void
 sentry__envelope_for_each_request(const sentry_envelope_t *envelope,
     bool (*callback)(sentry_prepared_http_request_t *,
         const sentry_envelope_t *, void *data),
     const sentry_rate_limiter_t *rl, void *data)
 {
     sentry_prepared_http_request_t *req;
-    sentry_uuid_t event_id = sentry__envelope_get_event_id(envelope);
 
     if (envelope->is_raw) {
-        req = prepare_http_request(&event_id, ENDPOINT_TYPE_ENVELOPE,
-            "application/x-sentry-envelope", envelope->contents.raw.payload,
+        req = prepare_http_request(envelope->contents.raw.payload,
```

Comment:
Would it make sense to change the function signature to just return a `sentry_prepared_http_request_t` now that there will only be one?

Reply:
Hm, I think we are pretty much guaranteed that we get ether 0 or 1 requests for a single envelope, right? In that case, it makes sense.

Reply:
I would say so. Theoretically, the SDK could also split up an envelope into multiple envelopes, but I wouldn't know what the use case would be. Rather combining multiple into one.